lib/logstash/outputs/http.rb

```diff
@@ -246,11 +227,11 @@ def send_event(event, attempt)
     body = event_body(event)
 
     # Send the request
-    url = event.sprintf(@url)
-    headers = event_headers(event)
+    url = @is_batch ? @url : event.sprintf(@url)
+    headers = @is_batch ? @headers : event_headers(event)
 
     # Compress the body and add appropriate header
-    if @http_compression == true
+    if @http_compression == true && !@is_batch
```

Comment:
I know this is to keep current behaviour but the [current behaviour is a bug](https://github.com/logstash-plugins/logstash-output-http/issues/87) so we can fix it here too, also replacing  https://github.com/logstash-plugins/logstash-output-http/pull/88